docs/immutable_tree.md

```diff
@@ -0,0 +1,69 @@
+# Immutable Tree
+
+### Structure
+
+The Immutable tree struct contains an IAVL version.
+
+```golang
+type ImmutableTree struct {
+	root    *Node
+	ndb     *nodeDB
+	version int64
+}
+```
+
+Using the root and the nodeDB, the ImmutableTree can retrieve any node that is a part of the IAVL tree at this version.
+
+Users can get information about the IAVL tree by calling getter functions such as `Size()` and `Height()` which will return the tree's size and height by querying the root node's size and height.
+
+### Get
+
+Users can get values by specifying the key or the index of the leaf node they want to get value for.
+
+Get by key will return both the index and the value.
+
+```golang
+// Get returns the index and value of the specified key if it exists, or nil
+// and the next index, if it doesn't.
+func (t *ImmutableTree) Get(key []byte) (index int64, value []byte) {
+	if t.root == nil {
+		return 0, nil
+	}
+	return t.root.get(t, key)
+}
+```
+
+Get by index will return both the key and the value. The index is the index in the sorted list of leaf nodes. The leftmost leaf has index 0. It's neighbor has index 1 and so on.
```

Comment:
I am a little confused about what the index is--how are the leaf nodes sorted, and is a sorted list stored somewhere? And also, why is it useful to be able to fetch a key/value by index?

Reply:
Nodes are lexicographically sorted by keys. This is what allows for efficient iteration over key ranges. This is one of the key properties of the IAVL tree.

Index is the absolute position of a given key.

Let me reread the spec and see how to make this clearer.

Reply:
Yes, the leaf nodes are sorted lexicographically in the tree itself. So getting index `0` will be the leftmost key, and so on. Though I haven't seen where `GetByIndex` is actually used in the code. Since `IterateRange` doesn't actually use `GetByIndex` in its implementation

Reply:
OK, cool. Thank you both!